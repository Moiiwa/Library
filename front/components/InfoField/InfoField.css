.inline {
    display: inline-block;
}

#infoFieldValue {
    margin-left: 0.5em;
}