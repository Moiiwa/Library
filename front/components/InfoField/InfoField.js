import React from 'react'
import './InfoField.css'

class InfoField extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                <p id='infoFieldLabel' className='inline'>{this.props.label}</p>
                <p id='infoFieldValue' className='inline'>{this.props.value}</p>
            </div>
        );
    }
}

export default InfoField;