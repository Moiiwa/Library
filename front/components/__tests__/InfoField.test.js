import React from 'react';
import InfoField from '../InfoField/InfoField';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('InfoField', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<InfoField />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('works with props', () => {
        const props = {
            label: 'Title',
            value: 'ValueTitle'
        }
        const wrapper = shallow(<InfoField {...props} />).find('#infoFieldLabel');
        expect(wrapper.text().includes('Title')).toBe(true);
    });

    it('works without props', () => {
        const props = {
            label: null,
            value: null
        }
        const wrapper = shallow(<InfoField {...props} />).find('.label');
        expect(wrapper).toEqual({});
    });
});