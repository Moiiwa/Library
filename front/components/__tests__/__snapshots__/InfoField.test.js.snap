// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoField renders correctly 1`] = `
<div>
  <p
    className="inline"
    id="infoFieldLabel"
  />
  <p
    className="inline"
    id="infoFieldValue"
  />
</div>
`;
